app/routes/admin+/_horses+/horses.edit.$horseId.tsx

```diff
@@ -80,6 +93,51 @@ export async function action({ request, params }: DataFunctionArgs) {
 		})
 	}
 
+	if (cooldown && cooldownStartDate && cooldownEndDate) {
+		// Get events horseID is registered for
+		const horseEvents = await prisma.event.findMany({
+			where: {
+				horses: {
+					some: {
+						id: updatedHorse.id,
+					},
+				},
+			},
+		})
+
+		// Compare event dates to cooldown dates and gather events with conflicts
+		const conflictEvents = []
+		if (horseEvents) {
+			for (const e of horseEvents) {
+				if (
+					cooldownStartDate <= e.start &&
+					e.start < add(cooldownEndDate, { days: 1 }) // checks that event is before midnight of cooldownEndDate
+				) {
+					conflictEvents.push(e)
+				}
+			}
+		}
+
+		// Remove horse from events with conflicts
+		if (conflictEvents.length > 0) {
+			for (const e of conflictEvents) {
+				await prisma.event.update({
+					where: { id: e.id },
+					data: {
+						horses: {
+							disconnect: { id: updatedHorse.id },
+						},
+					},
+				})
+			}
+			return json({
+				status: 'idle',
+				submission,
+				conflictEvents,
```

Comment:
Can you think of a better way of returning these events to be displayed on the page? Typescript is unhappy that the action is returning conflictEvents.

Reply:
I don't know if this is the best way to do this, but if you add a `conflictEvents: [],` field to the other return statements, that should resolve this particular type problem.